app/utils/ScientificSpecies/shouldUpdateSpeciesSync.ts

```diff
@@ -0,0 +1,53 @@
+import AsyncStorage from '@react-native-community/async-storage';
+import dbLog from '../../repositories/logs';
+import { LogTypes } from '../constants';
+
+// Checks if the gap between current date and the last species updated date is greater than 90 then returns true else returns false
+const shouldUpdateSpeciesSync = async (): Promise<boolean> => {
+  const currentDate = new Date().toISOString().split('T')[0];
+  const previousUpdatedSyncDate = await getSpeciesSyncUpdateDate();
+  console.log('previousUpdatedSyncDate', previousUpdatedSyncDate);
+
+  if (!previousUpdatedSyncDate) {
+    setSpeciesSyncUpdateDate(currentDate);
+  } else {
+    let differenceNoOfDays: any =
+      new Date(currentDate).getTime() - new Date(previousUpdatedSyncDate).getTime();
+
+    differenceNoOfDays = differenceNoOfDays * (1000 * 60 * 60 * 24);
```

Comment:
This should be rather `differenceNoOfDays / (1000 * 60 * 60 * 24); ` as otherwise the sync would be done every day!

Reply:
Fixed it. @ankitecd please check.